base/broadcast.jl

```diff
@@ -179,6 +179,17 @@ function Broadcasted{Style}(f::F, args::Args, axes=nothing) where {Style, F, Arg
     Broadcasted{Style, typeof(axes), Core.Typeof(f), Args}(f, args, axes)
 end
 
+andand(a, b) = a && b
+function broadcasted(::typeof(andand), a, bc::Broadcasted)
+    bcf = flatten(bc)
```

Comment:
Maybe I am missing something, but is the `flatten` here even needed?

Reply:
Yes.  If we simply did:

```julia
    return broadcasted((a, args...) -> a && bc.f(args...), a, bc.args...)
```

This would only go down "one layer" of laziness.  The way `Broadcasted`s work is that they recursively index into their arguments before evaluating their function at the current index.  Upon indexing into another `Broadcasted`, well, it'll index into its arguments _and then evaluate its function_ and then pass that result back up the chain.  We want to recursively move all broadcasteds on the RHS into an anonymous function — and that's _precisely_ what `flatten` does.

Reply:
Ah, I see. Makes sense